src/callbacks/scheduler.jl

```diff
@@ -63,17 +41,18 @@ function stateaccess(scheduler::Scheduler)
 end
 
 
-function init!(::Scheduler, learner)
+function init!(scheduler::Scheduler, learner)
     if !haskey(learner.cbstate, :hyperparams)
         learner.cbstate.hyperparams = MVHistory()
     end
+    scheduler.step = 1
 end
 
 
 function on(::StepBegin, phase::AbstractTrainingPhase, scheduler::Scheduler, learner)
```

Comment:
Does `StepBegin` correspond to the start of each mini-batch? If that's the case, then we might want to be specific about whether the "iteration unit" is in epochs or batches. ParameterSchedulers has `Interpolator` which allows specifying a schedule in epochs and then iterating in batches. `FluxTraining.Scheduler` could auto-wrap the schedule for convenience?

Reply:
`StepBegin` is the begin of a step/minibatch. Epochs are demaracated by `EpochBegin`/`EpochEnd`